api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -1172,6 +1172,58 @@ public void saveOrder_shouldSaveARevisedOrder() {
 		assertEquals(revisedOrder.getDateActivated(), DateUtils.addSeconds(originalOrder.getDateStopped(), 1));
 		assertFalse(originalOrder.isActive());
 	}
+	/**
+	 * @see OrderService#saveOrder(org.openmrs.Order, OrderContext)
+	 */
+	@Test
+	public void saveOrder_shouldSaveOrderWithFulfillerStatus() {
```

Comment:
I think this test isn't required

Reply:
Is there an existing test for testing saving an order?  I would just add setting and testing the fulfiller status there instead of creating a whole new test.

Reply:
Yes it does, and that's what I was thinking too.

Reply:
There is a saveOrder test that ensures that one can successfully place a new order, that's the one you need to update to set the fulfiller fields.

Reply:
@wluyima  The following test cases exist that create a new order and save it. I decided to add it to he bolded one. Feel free to suggest another one. But I do not in particular see a test case in OrderServiceTest that creates a order and tests if it was saved with all the according attributed

- saveOrder_shouldSetOrderTypeOfTestOrderToTestOrderIfNotSetAndConceptNotMapped
- saveOrder_shouldSetOrderTypeOfDrugOrderToDrugOrderIfNotSetAndConceptNotMappe
- **saveOrder_shouldPassForADiscontinuationOrderWithNoPreviousOrder**
- saveOrder_shouldSetConceptForDrugOrdersIfNull